src/wrappers/python/PyOpenEXR.cpp

```diff
@@ -467,7 +455,114 @@ PyPart::readDeepPixels(MultiPartInputFile& infile, const std::string& type, cons
+                    switch (C._type)
+                    {
+                      case UINT:
+                          S[y][x] = static_cast<void*>(new uint32_t[size]);
+                          break;
+                      case HALF:
+                          S[y][x] = static_cast<void*>(new half[size]);
+                        break;
+                      case FLOAT:
+                          S[y][x] = static_cast<void*>(new float[size]);
+                          break;
+                      default:
+                          throw std::runtime_error("invalid pixel type");
+                    } // switch c->type
+                }
+
+
+            std::cout << "clearing..." << std::endl;
+            
+            for (size_t y=0; y<height; y++)
+                for (size_t x=0; x<width; x++)
+                {
+                    auto size = sampleCount[y][x];
+                    switch (C._type)
+                    {
+                      case UINT:
+                          for (size_t i=0; i<size; i++)
+                          {
+                              auto s = static_cast<uint32_t*>(S[y][x]);
+                              s[i] = 0;
+                          }
+                          break;
+                      case HALF:
+                          for (size_t i=0; i<size; i++)
+                          {
+                              auto s = static_cast<half*>(S[y][x]);
+                              s[i] = 0;
+                          }
+                          break;
+                      case FLOAT:
+                          for (size_t i=0; i<size; i++)
+                          {
+                              auto s = static_cast<float*>(S[y][x]);
+                              s[i] = 0;
+                          }
+                          break;
+                      default:
+                          throw std::runtime_error("invalid pixel type");
+                    } // switch c->type
+                }
+        }
+        
+        std::cout << "part.readPixels..." << std::endl;
         part.readPixels (dw.min.y, dw.max.y);
+        std::cout << "part.readPixels...done." << std::endl;
+
+        for (auto c : channels)
+        {
+            auto C = c.second.cast<const PyChannel&>();
+            for (size_t y=0; y<height; y++)
+                for (size_t x=0; x<width; x++)
+                {
+                    auto size = sampleCount[y][x];
+                    auto &S = *C.deep_samples;
+                    switch (C._type)
+                    {
+                      case UINT:
+                          for (size_t i=0; i<size; i++)
+                          {
+                              auto s = static_cast<uint32_t*>(S[y][x]);
+                              std::cout << "sample: " << C.name << "[" << y << "][" << x << "][" << i << "]=" << s[i] << std::endl;
```

Comment:
std::endl flushes, in a loop like this, using `\n` and flushing at the end is probably infinitely faster, on windows, at least.

Reply:
Same here, the print statements are debugging in WIP code, I'll remove shortly, before merging.